drivers/platform/x86/thinkpad-wmi.c

```diff
@@ -278,40 +278,46 @@ struct thinkpad_wmi {
 };
 
 /* helpers */
-static int thinkpad_wmi_errstr_to_err(const char *errstr)
-{
-	if (!strcmp(errstr, "Success"))
-		return THINKPAD_WMI_SUCCESS;
-	if (!strcmp(errstr, "Not Supported"))
-		return THINKPAD_WMI_NOT_SUPPORTED;
-	if (!strcmp(errstr, "Invalid"))
-		return THINKPAD_WMI_INVALID;
-	if (!strcmp(errstr, "Access Denied"))
-		return THINKPAD_WMI_ACCESS_DENIED;
-	if (!strcmp(errstr, "System Busy"))
-		return THINKPAD_WMI_SYSTEM_BUSY;
-
-	pr_debug("Unknown error string: '%s'", errstr);
-
-	return -EINVAL;
-}
-
-static int thinkpad_wmi_extract_error(const struct acpi_buffer *output)
+static int thinkpad_wmi_check_output(const struct acpi_buffer *output,
+					      char **value)
 {
 	const union acpi_object *obj;
 	int ret;
 
 	obj = output->pointer;
 	if (!obj || obj->type != ACPI_TYPE_STRING || !obj->string.pointer)
-		return -EIO;
+		ret =  -EIO;
+
+	/* value assigned -> wmi call returning output */
+	else if(value) {
```

Comment:
Add braces if you have empty lines (see scripts/checkpatch.pl to automatically lint the code)

Reply:
yep, working on this

Reply:
done